logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/selfdeletingMessages/ObserveSelfDeletingMessagesUseCase.kt

```diff
@@ -32,7 +32,7 @@ interface ObserveSelfDeletingMessagesUseCase {
     suspend operator fun invoke(): Flow<SelfDeletingMessagesStatus>
 }
 
-class ObserveSelfDeletingMessagesUseCaseImpl internal constructor(
+class ObserveGlobalSelfDeletingMessagesUseCaseImpl internal constructor(
```

Comment:
I still find the name for this use case not really fitting to what it is doing 🤔

Reply:
ObserveSelfDeletingMessagesStatus ? :P

Reply:
Why would the app ever observe the global SelfDeleting setting?

Can't the UseCase be smarter and do it on a per-conversation level?

For example:
* Scenario A: Conversation has self-deleting timer on a conversation-level.
* Scenario B: Conversation has no self-deleting, but my team enforces a team-level restriction.
* Scenario C: Conversation has no self-deleting enforced, and my team has it completely disabled.
* Scenario D: Conversation has no self-deleting enforced, and my team has no restrictions on it.



If the app calls `observeConversationSelfDeletingSettings(conversation)` it should get something like this:

```kotlin
sealed interface ConversationSelfDeletingTimer {
    object Disabled: ConversationSelfDeletingTimer
    object Enabled: ConversationSelfDeletingTimer
    data class Enforced(val duration: Duration): ConversationSelfDeletingTimer
}
```

Now... based on this, the UI can do the following, when its value is:
- Enabled: Let the user choose a timer, as there are no restrictions.
- Disabled: Block the user from using the timer, as it was disabled by the group or team they belong to.
- Enforced: Show the user a specific timer that was set by the group or by the team they belong to.

Reply:
Yes. I totally failed understanding that the conversations also had their own self deleting timer that should be combined with the team settings to get the final state. But the way you summed it up is really good. So I will proceed with that instead and add the changes as well on the Kalium layer to be able to persist these settings per conversation.